sim/core/attack.go

```diff
@@ -187,15 +205,27 @@ func (ahe *SpellHitEffect) calculateDamage(sim *Simulation, ability *SimpleSpell
 		dmg += ahe.WeaponInput.CalculateDamage(attackPower, bonusWeaponDamage)
 	} else if ahe.WeaponInput.DamageMultiplier != 0 {
 		// Bonus weapon damage applies after OH penalty: https://www.youtube.com/watch?v=bwCIU87hqTs
-		if ability.OutcomeRollCategory.Matches(OutcomeRollCategoryRanged) {
-			dmg += character.AutoAttacks.Ranged.calculateSwingDamage(sim, attackPower) + bonusWeaponDamage
-		} else if ahe.IsMH() {
-			dmg += character.AutoAttacks.MH.calculateSwingDamage(sim, attackPower) + bonusWeaponDamage
+		// TODO not all weapon damage based attacks "scale" with +bonusWeaponDamage (e.g. Devastate, Shiv, Mutilate don't)
+		if ahe.WeaponInput.Normalized {
+			if ability.OutcomeRollCategory.Matches(OutcomeRollCategoryRanged) {
+				dmg += character.AutoAttacks.Ranged.calculateNormalizedWeaponDamage(sim, attackPower) + bonusWeaponDamage
+			} else if ahe.IsMH() {
+				dmg += character.AutoAttacks.MH.calculateNormalizedWeaponDamage(sim, attackPower) + bonusWeaponDamage
+			} else {
+				dmg += character.AutoAttacks.OH.calculateNormalizedWeaponDamage(sim, attackPower)*0.5 + bonusWeaponDamage
+			}
 		} else {
-			dmg += character.AutoAttacks.OH.calculateSwingDamage(sim, attackPower)*0.5 + bonusWeaponDamage
+			if ability.OutcomeRollCategory.Matches(OutcomeRollCategoryRanged) {
+				dmg += character.AutoAttacks.Ranged.calculateWeaponDamage(sim, attackPower) + bonusWeaponDamage
+			} else if ahe.IsMH() {
+				dmg += character.AutoAttacks.MH.calculateWeaponDamage(sim, attackPower) + bonusWeaponDamage
+			} else {
+				dmg += character.AutoAttacks.OH.calculateWeaponDamage(sim, attackPower)*0.5 + bonusWeaponDamage
+			}
```

Comment:
Instead of repeating this block twice, it might be cleaner to add a 'normalized' parameter to calculateWeaponDamage so that it can handle both normalized and unnormalized attacks.

Reply:
This is kept "dumb" intentionally, since there's quite a bit more stuff that needs to be handled here, or around here (attackerBonusAttackPower, bonusDamage, possibly bonusDamageDone)

Reply:
Ok, repeating code is always bad but we can come back to this later once the full picture is in place.